### PR TITLE
Fix: fix project submissions' page overflow bug.

### PR DESCRIPTION
**Pull request by IAmYoungbossy**
## Because

This pull request fixes the page overflow issue on The Odin Project's project_submissions page. Currently, when users click on the "View full list of solutions" link on any project page ( eg. [View full list of solutions](https://www.theodinproject.com/lessons/596/project_submissions) ), the resulting page layout is broken on mobile devices and smaller screen sizes. The issue is caused by two main problems: the hint--top class from the Hint.css library and the pagination's fixed width as shown below with some screenshots.

1.  The hint--top class from Hint.css library was contributing to the page overflow. Below is a screenshot to demostrate my claim. 

#### Bug: Overflow due to hint--top class from Hint.css library - Screenshot
bug-caused-by-hint-css-library

2. The pagination is not responsive and also results to page overflow. Below is the screenshot.

#### Bug: Overflow due to fixed with on pagination - Screenshot
bug-page-overflow

After fixing the above below are the screenshot of the results;

1. Change hint--top class to hint--top-left.

#### Fix - Screenshot
fix-page-overflow3

2. Add overflow-x: auto; rule to nav pagination container. This makes pagination nav links scrollable on smaller devices.

#### Fix - Screenshot
fix-page-overflow2

## This PR

- [Replaced the justify-center utility class with the mx-auto and w-fit classes to ensure the container is centered and the width adjusts to the content](https://github.com/TheOdinProject/theodinproject/commit/0b22f899de27dec30f75bf06c41ba2c49d413f02)

- [Added the overflow-x: auto rule to the pagy_nav container div element to make the pagination nav links scrollable on smaller devices.](https://github.com/TheOdinProject/theodinproject/commit/dbef0c9d41eaf93823959ce9d60a3dc0ece036a9)

- [Changed the hint.css library class from hint--top to hint--top-left to prevent the hint from overflowing the container.](https://github.com/TheOdinProject/theodinproject/commit/7ea1a443d9c1c2aeff3deafc949163ffd033174d)

## Issue

Closes #XXXXX

## Additional Information

After the fixes the project_submissions' pages becomes fully responsive on smaller screen devices as well as bigger screen devices.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

File: app/views/lessons/project_submissions/index.html.erb [hunk 1]
```diff
@@ -17,7 +17,7 @@
           userSubmission: @user_submission
         }
       ) %>
-  <div class="text-center">
+  <div class="text-center" style="overflow-x: auto;">
```
Comment (KevinMulhern) [inline app/views/lessons/project_submissions/index.html.erb:20]: Nit: Can we use the tailwind class for this please?
```suggestion
  <div class="text-center overflow-x-auto">
```

**Review by KevinMulhern — CHANGES_REQUESTED** [review]
Looks great, thanks for fixing this @IAmYoungbossy. Just one minor suggestion from me.

File: app/views/lessons/project_submissions/index.html.erb (hunk 1 above)
Comment (IAmYoungbossy) [reply app/views/lessons/project_submissions/index.html.erb:20]: Yes, I'll do that right away. Thanks.

**Review by KevinMulhern — APPROVED** [review]
Great stuff, thank you @IAmYoungbossy

**Comment by IAmYoungbossy** [comment]
You're welcome @KevinMulhern it's a pleasure.

**Pull request merged**